text/0000-location.md

```diff
@@ -0,0 +1,440 @@
+
+|Master Data Common Name|Attribute Name|Description|Example|
+|-----------------------|--------------|-----------|-------|
+|Location Identifier|location_id|A location is referenced using a location_id (key attribute). For GS1 locations, the location_id is a Global Location Number which is part of GS1 specifications.|"0099474000005"|
+|Location Namespace|location_namespace|This RFC defines a single location_namespace for GS1; a location in the GS1 location_namespace is called a GS1 location. Note that this design supports extending additional location namespaces in the future.|"01"|
+|Owner|owner/org_id|The identifier of the organization responsible for maintaining the location. For GS1 locations, org_id is the Pike organization which claims ownership of the GS1 company prefix (via the "gs1_company_prefixes" field in the Pike records). The GLN will always start with the company prefix. The smart contract will check that the org claiming ownership lists the matching prefix in their Pike org.||
+
+
+## Transactions
+[transactions]: #transactions
+
+In order to add Locations to state, a transaction must be used. The following 
+transactions and their execution rules are designed for use with Hyperledger
+Sawtooth's Sabre smart contract engine running via Hyperledger Transact, which
+covers all currently implemented DLT backends in Grid.
+
+Locations are managed by submitting transactions to Hyperledger Grid, which will
+process them with the Grid Location smart contract. The following transactions 
+are supported:
+- `LocationCreateAction` - create a Location and store it in state.
+- `LocationUpdateAction` - update (replace) the properties of a Location already in state.
+- `LocationDeleteAction` - remove a Location from state.
+ 
+_*Disclaimer: `LocationDeleteAction` is a potentially hazardous operation and needs to 
+be done with care._
+
+
+## Permissions
+[permissions]: #permissions
+
+Creation of GS1 locations is restricted to agents acting on behalf of the 
+organization that is associated with the GS1 company prefix encoded in the GLN. 
+Like Grid Product, an organization must have the GS1 company prefix in the 
+“gs1_company_prefixes” metadata field for its Pike organization. When a 
+location is created, its owning organization is stored with the location in an 
+“owner” field.
+
+Updates to locations are restricted to agents acting on behalf of the 
+organization stored in the location’s owner field. Only property fields can be 
+updated. They are updated by providing a complete and updated list of the 
+properties, and overwrites the entire list of property values. Location_id, 
+location_namespace, and owner fields are immutable.
+
+Deletion of locations is restricted to agents acting on behalf of the 
+organization in the location’s owner field. A setting will turn off deletion 
+entirely, with the intent that it could be enabled/disabled by a Grid 
+administrator. Disabling delete is useful because external systems may have 
```

Comment:
The wording leads me into a couple of scenarios and questions.

- Is the setting and turning off delete entirely the right option? Would a setting for the entity make more sense? I like the by entity to the setting that AWS has in EC2 that allows `termination protection` so you can't accidentally delete a "stack" without a bunch of clicks. A global setting feels easier to implement (global bool) but doesn't provide some flexibility. That then leads addressing how deletes take place? I ask a Grid admin? Does the Grid admin implement a `--force` CLI switch? Turn the global flag off and open up the window to allow non-admins to retry a failed delete and then blow away something else.
- Assume the original flag is used ... the RFC doesn't seem to address how a delete happens. The Grid admin can set a global flag. But the `allow_delete` flag, it's config, and the Grid admin scope of actions isn't called out.
- Can a workflow of setting `deactivated` be set prior to deleting? this would allow transactions to complete and safeguard a delete

Reply:
> I'm still hung up a little on the workflow for a location and `status`. xref [#20 (comment)](https://github.com/hyperledger/grid-rfcs/pull/20#discussion_r478818306)
> 
> One of the use cases here is still having the owning company wanting to put a location into maintenance, like modernizing a dock space but not wanting to remove the whole warehouse from being available. Is there some other attribute/method that we'd use? This might be covered by some other data set attribute from GS1 but there is an absolute when it comes to supply chain ... damage happens at docks, new tech is introduced, and/or the warehouse gear craps out. Should the location RFC account for planned and unplanned `do not use` status updates to a location that could be a space like a dock?
> 
> If the current scope is a higher level business decision, like activating or termination of warehouse space lease - cooool. I will move on and back the future RFC approach. I just want to make sure the RFC states this and doesn't leave too many interpretations of "active" to a location.

i feel like the context of the Location status will trend toward "but not for everyone!" and would suggest that type of attribution should probably fall to a separate [tbd] Grid capability.  The nuances here are gonna end-up at 32 flavors and then some.

Reply:
I don’t think that covers the question on what Grid should enable to handle a status of a Location. There’s already Create and Inactivate. 

I’ll resolve this portion and open an issue/RFC if more discussion is needed, as well as :rocket: chat.

Reply:
I don’t think that covers the question on what Grid should enable to handle a status of a Location. There’s already Create and Inactivate. 

I’ll resolve this portion and open an issue/RFC if more discussion is needed, as well as :rocket: chat.